sphinx/environment/__init__.py

```diff
@@ -502,10 +502,35 @@ def get_outdated_files(self, config_changed: bool) -> tuple[set[str], set[str],
                         # this will do the right thing when dep is absolute too
                         deppath = path.join(self.srcdir, dep)
                         if not path.isfile(deppath):
+                            logger.debug(
+                                '[build target] changed %r '
+                                'missing dependency %r',
+                                docname,
+                                deppath)
                             changed.add(docname)
                             break
                         depmtime = _last_modified_time(deppath)
                         if depmtime > mtime:
+                            try:
+                                depmtime_dt = datetime.utcfromtimestamp(depmtime)
+                            except ValueError:  # e.g., year 53606865 is out of range
```

Comment:
Honestly, I think this is unnecessary because it's likely that when we will hit this year, `datetime.utcfromtimestamp` would handle it correctly. Also, it complicates the flow and unnecessarily introduces a `try-and-catch` block which is costly in general.

Reply:
In addition, I think that we report the time in UTC (somewhere else in the file).

Reply:
I only added the try/except because CIs complained. I'll revert it so we can see the failure again (I rebased so can't find it easily anymore)

Reply:
Wait, what do you mean by CIs complained? did it complained because of `utcfromtimestamp`?

Reply:
> Wait, what do you mean by CIs complained? 

See the main comment thread, I accidentally replied there to this bit.

> try-and-catch block which is costly in general.

Can you explain whaht you mean? I haven't experienced slowdowns at least. `timeit` seems to think it's a negligible cost:
```
$ python -m timeit -s "import datetime" "datetime.datetime.fromtimestamp(1000.)"
500000 loops, best of 5: 414 nsec per loop
$ python -m timeit -s "import datetime" "try:" "    datetime.datetime.fromtimestamp(1000.)" "except:" "    pass"
500000 loops, best of 5: 417 nsec per loop
```

Reply:
Ok, maybe my knowledge was a bit too old. Since Python 3.11, there are 0-cost exceptions so it should be fine. It's just that prior to 3.11, a try-catch would make the frame larger. As such, I'm not sure how important is the effect when Sphinx is used for *very* large projects (when there are possibly million of files to be processed). 

Also, prior to 3.11, I think that Python may have already have a somewhat effective try-catch in case there is no exception being thrown. 

TL;DR: I was wrong!